app/code/Magento/CompareListGraphQl/Model/Service/CustomerService.php

```diff
@@ -0,0 +1,158 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CompareListGraphQl\Model\Service;
+
+use Magento\Catalog\Model\CompareList;
+use Magento\Catalog\Model\CompareListFactory;
+use Magento\Catalog\Model\ResourceModel\Product\Compare\CompareList as ResourceCompareList;
+use Magento\Customer\Api\AccountManagementInterface;
+use Magento\Customer\Api\CustomerRepositoryInterface;
+use Magento\Customer\Model\AuthenticationInterface;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\GraphQl\Exception\GraphQlAuthenticationException;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Exception\GraphQlNoSuchEntityException;
+
+/**
+ * Service class for customer
+ */
+class CustomerService
```

Comment:
I would strongly suggest splitting this class into 3 different classes:
- GetListIdByCustomerId
- SetCustomerToCompareList
- ValidateCustomer

Reply:
**Done**